test/test_device.py

```diff
@@ -202,6 +203,56 @@ def setUp(self):
         if '--testnet' not in self.dev_args:
             self.dev_args.append('--testnet')
 
+    def _generate_and_finalize(self, unknown_inputs, psbt):
+        if not unknown_inputs:
+            # Just do the normal signing process to test "all inputs" case
+            sign_res = process_commands(self.dev_args + ['signtx', psbt['psbt']])
+            finalize_res = self.wrpc.finalizepsbt(sign_res['psbt'])
+        else:
+            # Sign only input one on first pass
+            # then rest on second pass to test ability to successfully
+            # ignore inputs that are not its own. Then combine both
+            # signing passes to ensure they are actually properly being
+            # partially signed at each step.
+            first_psbt = PSBT()
+            first_psbt.deserialize(psbt['psbt'])
+            second_psbt = PSBT()
+            second_psbt.deserialize(psbt['psbt'])
+
+
+            # Blank master fingerprint to make hww fail to sign
+            # Single input PSBTs will be fully signed by first signer
+            for psbt_input in first_psbt.inputs[1:]:
+                for pubkey, path in psbt_input.hd_keypaths.items():
+                    psbt_input.hd_keypaths[pubkey] = (0,0,0,0)+path[4:]
```

Comment:
The master fingerprint is not a tuple of bytes. It's an integer that is in `path[0]`. So this is wrong and not the correct way to blank out the master fingerprint. It should really be `psbt_input.hd_keypaths[pubkey] = [0]+path[1:]`

Reply:
the keypath itself is a tuple of integers, so it's actually:

`psbt_input.hd_keypaths[pubkey] = (0,)+path[1:]`

to allow concatenation.

Reply:
fixed